app/controllers/user/sessions_controller.rb

```diff
@@ -0,0 +1,47 @@
+class User::SessionsController < Devise::SessionsController
+  def create
```

Comment:
override the `#new` method where `:user_sign_in_uid` is deleted from the session (and `super` is called) to avoid forcing entry of the OTP key for a different user during sign ins; also adapt the `routes.rb` for that